iib/workers/dogpile_cache.py

```diff
@@ -0,0 +1,63 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import functools
+
+from dogpile.cache import make_region
+
+from iib.workers.config import get_worker_config
+
+
+def should_use_cache(*args):
+    """Return true in case this requests can be taken from or stored in cache."""
+    return any([arg.find('@sha256:') != -1 for arg in args])
+
+
+def skopeo_cache(dogpile_region):
```

Comment:
I think you could make this a generic decorator which accepts a parameter for a function that analyzes the arguments and determines if the cache should be used.

Reply:
Hi @mprahl, I changed the decorator to be more general. Could you check it if it is implemented in a way you suggested. 
Thank you, Jan